packages/lesswrong/components/form-components/FMCrosspostControl.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState, useEffect } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import {
+  fmCrosspostSiteNameSetting,
+  fmCrosspostBaseUrlSetting,
+} from "../../lib/instanceSettings";
+import { useSingle } from "../../lib/crud/withSingle";
+import { useCrosspostApolloClient } from "../hooks/useCrosspostApolloClient";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import LoginIcon from "@material-ui/icons/LockOpen"
+import UnlinkIcon from "@material-ui/icons/RemoveCircle";
+import classNames from "classnames";
+import { gql, useMutation } from "@apollo/client";
+import { useCurrentUser } from "../common/withUser";
+import { useOnTabView } from "../hooks/useOnTabView";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "flex",
+    flexDirection: "column",
+    margin: 8,
+  },
+  link: {
+    color: theme.palette.primary.main,
+  },
+  crosspostMessage: {
+    marginBottom: 12,
+  },
+  button: {
+    background: theme.palette.buttons.imageUpload.background,
+    "&:hover": {
+      background: theme.palette.buttons.imageUpload.hoverBackground,
+    },
+    color: theme.palette.text.invertedBackgroundText,
+  },
+  buttonIcon: {
+    fontSize: 18,
+    marginRight: theme.spacing.unit,
+  },
+  error: {
+    color: theme.palette.error.main,
+    margin: "8px 0",
+  },
+});
+
+const FMCrosspostAccount = ({fmCrosspostUserId, classes}: {
```

Comment:
I'd personally choose to write out "forum" rather than "fm". Not that many characters saved, but then my brain doesn't have to do extra parsing (also generally we don't drop letters as a naming convention)

Reply:
Reading through the code more, I think `fm` is kind of confusing/jarring and should just be `forum`